### DISCUSSION TITLE
Creating a public link

### DISCUSSION THREAD

**Discussion by filsanet** [lllyasviel/Fooocus#6] [General]
After starting the run.bat file (Windows), the console prints some text

```To create a public link, set `share=True` in `launch()`.```

However it is not clear where to do this. Would be helpful to have some commented out code somewhere that supports this. 

Really nice job with Fooocus! I think it is fantastic so far.

**Comment by dayunbao** [comment]
That message is from the Automatic1111 web UI, I believe.  Check out it's repo to learn more.

**Comment by filsanet** [comment]
Ok, I figured it out. 

I edited `run.bat` , and added "--listen" to the end of the command.

```
.\python_embeded\python.exe -s Fooocus\entry_with_update.py --listen
```

If it works, you will notice the URL of the message change to 0.0.0.0:

* before (not public): `Running on local URL:  http://127.0.0.1:7860`
* after ( public ): `Running on local URL:  http://0.0.0.0:7860`

There's a comment in the README.md that tipped me off. However the README.md is slightly out of date.

**Reply by unpredictable1996** [reply]
hey, made a comment below, working on 2.1.851

**Comment by xx88xx** [comment]
did you ever find a way to make the link public? I also successfully got the 0.0.0.0 link, but not sure what do from here.

**Reply by unpredictable1996** [reply]
hey, made a comment below, working on 2.1.851

**Comment by unpredictable1996** [comment]
hey guys, just edited some code from a discussion thread here
Open Fooocus folder/webui.py (for ex in IDLE), check the last paragraph
It should be

```
shared.gradio_root.launch(
inbrowser=args_manager.args.in_browser,
server_name=args_manager.args.listen,
server_port=args_manager.args.port,
share=True,
auth=check_auth if args_manager.args.share and auth_enabled else None,
blocked_paths=[constants.AUTH_FILENAME]
)
```
![8 071421](https://github.com/lllyasviel/Fooocus/assets/34282583/ac348244-eb1b-42e1-bdfc-3049af9ccadf)

**Comment by lllyasviel** [comment]
just run with --share

**Reply by unpredictable1996** [reply]
Hi, it seems that just adding argument --share is not working for me
![image](https://github.com/lllyasviel/Fooocus/assets/34282583/f583fbd4-b02d-43b3-af52-5e18e3e9b4f0)
As it works with automatic1111, yep.
But adding share=True in web-ui.py solves it. Anyway looking forward to sharing as in a1111 in the future, maybe something works another way here

**Comment by vicento** [comment]
python entry_with_update.py  --share
